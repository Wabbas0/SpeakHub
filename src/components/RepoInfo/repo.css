.repo__text {
  font-size: 1.4em;
}

.repo__tag {
  padding: 2px 5px;
  border-radius: 12px;
}

#repo .repo__badge {
  background: #777;
}
