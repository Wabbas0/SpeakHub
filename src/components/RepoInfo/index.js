// @flow
import React from "react";
import { Breadcrumb, Skeleton, Badge, Typography, Tag } from "antd";

import "./repo.css";

type Props = {
  issuesCount: number,
  pullsCount: number,
  repoName: string,
  repoOwner: string,
  loading: boolean
};
const RepoInfo = ({
  issuesCount,
  pullsCount,
  repoName,
  repoOwner,
  loading
}: Props) => (
  <Skeleton
    active
    title={false}
    loading={loading}
    paragraph={{ rows: 2, width: [200, 150] }}
  >
    <div className="repo">
      <Breadcrumb>
        <Breadcrumb.Item>
          <Typography.Text className="repo__text" strong>
            {repoOwner}
          </Typography.Text>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <Typography.Text className="repo__text" strong>
            {repoName}
          </Typography.Text>
        </Breadcrumb.Item>
      </Breadcrumb>

      <div>
        <Tag className={"repo__tag"}>
          <Typography.Text>
            Issues{" "}
            <Badge
              className="repo__badge"
              count={issuesCount}
              style={{ background: "#777" }}
            />
          </Typography.Text>
        </Tag>

        <Tag className={"repo__tag"}>
          <Typography.Text>
            Pulls{" "}
            <Badge
              className="repo__badge"
              count={pullsCount}
              style={{ background: "#777" }}
            />
          </Typography.Text>
        </Tag>
      </div>
    </div>
  </Skeleton>
);

export default RepoInfo;
