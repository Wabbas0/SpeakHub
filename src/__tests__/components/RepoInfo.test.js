import React from "react";
import { create } from "react-test-renderer";
import RepoInfo from "../../components/RepoInfo";

describe("RepoInfo component", () => {
  test("it matches the snapshot when passing all required props", () => {
    const component = create(
      <RepoInfo
        repoName="react"
        repoOwner="facebook"
        issuesCount={10}
        pullsCount={20}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
