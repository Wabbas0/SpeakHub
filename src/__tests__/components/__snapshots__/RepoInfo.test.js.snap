// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoInfo component it matches the snapshot when passing all required props 1`] = `
<div
  className="repo"
>
  <div
    className="ant-breadcrumb"
  >
    <span>
      <span
        className="ant-breadcrumb-link"
      >
        <span
          aria-label={null}
          className="ant-typography repo__text"
          style={
            Object {
              "WebkitLineClamp": null,
            }
          }
        >
          <strong>
            facebook
          </strong>
        </span>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
    <span>
      <span
        className="ant-breadcrumb-link"
      >
        <span
          aria-label={null}
          className="ant-typography repo__text"
          style={
            Object {
              "WebkitLineClamp": null,
            }
          }
        >
          <strong>
            react
          </strong>
        </span>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        /
      </span>
    </span>
  </div>
  <div>
    <div
      className="ant-tag repo__tag"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span
        aria-label={null}
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        Issues
         
        <span
          className="repo__badge ant-badge ant-badge-not-a-wrapper"
        >
          <sup
            className="ant-scroll-number ant-badge-count ant-badge-multiple-words"
            data-show={true}
            style={
              Object {
                "background": "#777",
              }
            }
            title={10}
          >
            <span
              className="ant-scroll-number-only"
              style={
                Object {
                  "WebkitTransform": "translateY(-1100%)",
                  "msTransform": "translateY(-1100%)",
                  "transform": "translateY(-1100%)",
                  "transition": "none",
                }
              }
            >
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className=""
              >
                0
              </p>
              <p
                className="current"
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
            </span>
            <span
              className="ant-scroll-number-only"
              style={
                Object {
                  "WebkitTransform": "translateY(-1000%)",
                  "msTransform": "translateY(-1000%)",
                  "transform": "translateY(-1000%)",
                  "transition": "none",
                }
              }
            >
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className="current"
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
            </span>
          </sup>
        </span>
      </span>
    </div>
    <div
      className="ant-tag repo__tag"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <span
        aria-label={null}
        className="ant-typography"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        Pulls
         
        <span
          className="repo__badge ant-badge ant-badge-not-a-wrapper"
        >
          <sup
            className="ant-scroll-number ant-badge-count ant-badge-multiple-words"
            data-show={true}
            style={
              Object {
                "background": "#777",
              }
            }
            title={20}
          >
            <span
              className="ant-scroll-number-only"
              style={
                Object {
                  "WebkitTransform": "translateY(-1200%)",
                  "msTransform": "translateY(-1200%)",
                  "transform": "translateY(-1200%)",
                  "transition": "none",
                }
              }
            >
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className="current"
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
            </span>
            <span
              className="ant-scroll-number-only"
              style={
                Object {
                  "WebkitTransform": "translateY(-1000%)",
                  "msTransform": "translateY(-1000%)",
                  "transform": "translateY(-1000%)",
                  "transition": "none",
                }
              }
            >
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className="current"
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
              <p
                className=""
              >
                0
              </p>
              <p
                className=""
              >
                1
              </p>
              <p
                className=""
              >
                2
              </p>
              <p
                className=""
              >
                3
              </p>
              <p
                className=""
              >
                4
              </p>
              <p
                className=""
              >
                5
              </p>
              <p
                className=""
              >
                6
              </p>
              <p
                className=""
              >
                7
              </p>
              <p
                className=""
              >
                8
              </p>
              <p
                className=""
              >
                9
              </p>
            </span>
          </sup>
        </span>
      </span>
    </div>
  </div>
</div>
`;
